ch.sbb.scion.rcp.microfrontend/src/ch/sbb/scion/rcp/microfrontend/RouterOutlet.java

```diff
@@ -78,25 +78,26 @@ public RouterOutlet(final Composite parent, final int style, final String outlet
 
     routerOutletProxy = new RouterOutletProxy(outletName);
 
-    browser = new Browser(this, SWT.EDGE);
-    browser.addProgressListener(new ProgressAdapter() {
+    browser = BrowserViewFactory.createBrowserView(BrowserViewType.JXBROWSER, this);
+    browser.addNavigationListener(new NavigationListener() {
 
       private final List<IDisposable> disposables = new ArrayList<>();
 
       @Override
-      public void completed(final ProgressEvent event) {
+      public void onFrameLoadFinished() {
         // is invoked when completed loading the app, or when reloading it, e.g., due to
         // hot code replacement during development
         disposables.forEach(IDisposable::dispose);
         disposables.clear();
 
-        browser.execute(Resources.readString("js/helpers.js"));
+        browser.executeJavaScript(Resources.readString("js/helpers.js"));

```

Comment:
Let's extract
```
Resources.readString("js/helpers.js")
```
into a variable. There are at least two benefits:
* debuggability: observe the value of `Resources.readString("js/helpers.js")` in a variable,
* reading the potential exception stack trace: there may be several very different exceptions in this line.